code/command_block/command_block.py

```diff
@@ -0,0 +1,70 @@
+import os
+from time import sleep
+import subprocess
+import configparser
+
+lastLine = None
+sensor_logs = "foo"
+text_buffer = "buffer.txt"
+lora_status_file = "lora_status.txt"
+rx_file = "foo/rx.txt"
+
+def image_request(line):
+    camera_log = open(sensor_logs + "camera.txt")
+    lora_status = open(lora_status_file, "a")
+    tstamps = camera_log.readlines()
+    t = int(line[1:6])
+    q = line[6]
+    ti = min([abs(t-int(x)) for x in tstamps])
+    tim = t - ti if str(t - ti)+"\n" in tstamps else t + ti
+    lora_status.write("I " + str(tim) + " " + str(q) + '\n')
+    lora_status.close()
+
+def thermal_request(line):
+    thermal_log = open(sensor_logs + "thermal.txt")
+    lora_status = open(lora_status_file, "a")
+    tstamps = thermal_log.readlines()
+    t = int(line[1:6])
+    ti = min([abs(t-int(x)) for x in tstamps])
+    tim = t - ti if str(t - ti)+"\n" in tstamps else t + ti
+    lora_status.write("T " + str(tim) + " " + 0 + '\n')
+    lora_status.close()
+
+def system_reboot():
+    cmd = "reboot"
+    print("rebooting")
+    subprocess.Popen(cmd)
+
+def send_system_info():
+    lora_status = open(lora_status_file, "a")
+    lora_status.write("S 0 0")
+    lora_status.close()
+
+def append_text(line):
+    text = str(line[1:10])
+    buffer = open(text_buffer, "a")
+    buffer.write(text)
+
+def clear_buffer():
+    open(text_buffer,'w').close()
+
+while True:
```

Comment:
The loop should prevent reading the same command twice. Maybe consider monitoring differences in file length? (works perfectly fine in LoRa block)

Reply:
You mean size of the input file with packets - so when a new packet arrives it’s size changes and we know that we have a new packet. Am I right?

Reply:
I believe it's done in df5d08f9ec573d5e8aae144c7db96a12f2e98dc3